src/qt/optionsmodel.cpp

```diff
@@ -335,260 +335,272 @@ QVariant OptionsModel::data(const QModelIndex & index, int role) const
-                setRestartRequired(true);
-            }
-            break;
-        case Listen:
-            if (settings.value("fListen") != value) {
-                settings.setValue("fListen", value);
-                setRestartRequired(true);
-            }
-            break;
-        case Server:
-            if (settings.value("server") != value) {
-                settings.setValue("server", value);
-                setRestartRequired(true);
-            }
-            break;
-        default:
-            break;
+    case DisplayUnit:
+        setDisplayUnit(value);
+        break;
+    case ThirdPartyTxUrls:
+        if (strThirdPartyTxUrls != value.toString()) {
+            strThirdPartyTxUrls = value.toString();
+            settings.setValue("strThirdPartyTxUrls", strThirdPartyTxUrls);
+            setRestartRequired(true);
         }
+        break;
+    case Language:
+        if (settings.value("language") != value) {
+            settings.setValue("language", value);
+            setRestartRequired(true);
+        }
+        break;
+    case UseEmbeddedMonospacedFont:
+        m_use_embedded_monospaced_font = value.toBool();
+        settings.setValue("UseEmbeddedMonospacedFont", m_use_embedded_monospaced_font);
+        Q_EMIT useEmbeddedMonospacedFontChanged(m_use_embedded_monospaced_font);
+        break;
+    case CoinControlFeatures:
+        fCoinControlFeatures = value.toBool();
+        settings.setValue("fCoinControlFeatures", fCoinControlFeatures);
+        Q_EMIT coinControlFeaturesChanged(fCoinControlFeatures);
+        break;
+    case EnablePSBTControls:
+        m_enable_psbt_controls = value.toBool();
+        settings.setValue("enable_psbt_controls", m_enable_psbt_controls);
+        break;
+    case Prune:
+        if (settings.value("bPrune") != value) {
+            settings.setValue("bPrune", value);
+            setRestartRequired(true);
+        }
+        break;
+    case PruneSize:
+        if (settings.value("nPruneSize") != value) {
+            settings.setValue("nPruneSize", value);
+            setRestartRequired(true);
+        }
+        break;
+    case DatabaseCache:
+        if (settings.value("nDatabaseCache") != value) {
+            settings.setValue("nDatabaseCache", value);
+            setRestartRequired(true);
+        }
+        break;
+    case ThreadsScriptVerif:
+        if (settings.value("nThreadsScriptVerif") != value) {
+            settings.setValue("nThreadsScriptVerif", value);
+            setRestartRequired(true);
+        }
+        break;
+    case Listen:
+        if (settings.value("fListen") != value) {
+            settings.setValue("fListen", value);
+            setRestartRequired(true);
+        }
+        break;
+    case Server:
+        if (settings.value("server") != value) {
+            settings.setValue("server", value);
+            setRestartRequired(true);
+        }
+        break;
+    default:
+        break;
```

Comment:
I wonder if it should set `successful = false`? Or maybe just drop the `default` cases and let the compiler complain about unhandled cases?